concepts/severityscores/sofa.sql

```diff
@@ -0,0 +1,230 @@
+-- ------------------------------------------------------------------
+-- Title: Sequential Organ Failure Assessment (SOFA)
+-- This query extracts the sequential organ failure assessment (formally: sepsis-related organ failure assessment).
+-- This score is a measure of organ failure for patients in the ICU.
+-- The score is calculated on the first day of each ICU patients' stay.
+-- ------------------------------------------------------------------
+
+-- Reference for SOFA:
+--    Jean-Louis Vincent, Rui Moreno, Jukka Takala, Sheila Willatts, Arnaldo De Mendonça,
+--    Hajo Bruining, C. K. Reinhart, Peter M Suter, and L. G. Thijs.
+--    "The SOFA (Sepsis-related Organ Failure Assessment) score to describe organ dysfunction/failure."
+--    Intensive care medicine 22, no. 7 (1996): 707-710.
+
+-- Variables used in SOFA:
+--  GCS, MAP, FiO2, Ventilation status (sourced from CHARTEVENTS)
+--  Creatinine, Bilirubin, FiO2, PaO2, Platelets (sourced from LABEVENTS)
+--  Dobutamine, Epinephrine, Norepinephrine (sourced from INPUTEVENTS_MV and INPUTEVENTS_CV)
+--  Urine output (sourced from OUTPUTEVENTS)
+
+-- The following views required to run this query:
+--  1) uofirstday - generated by urine-output-first-day.sql
+--  2) vitalsfirstday - generated by vitals-first-day.sql
+--  3) gcsfirstday - generated by gcs-first-day.sql
+--  4) labsfirstday - generated by labs-first-day.sql
+--  5) bloodgasfirstdayarterial - generated by blood-gas-first-day-arterial.sql
+--  6) echodata - generated by echo-data.sql
+--  7) ventdurations - generated by ventilation-durations.sql
+
+-- Note:
+--  The score is calculated for *all* ICU patients, with the assumption that the user will subselect appropriate STAY_IDs.
+--  For example, the score is calculated for neonates, but it is likely inappropriate to actually use the score values for these patients.
+
+DROP TABLE IF EXISTS `physionet-data.mimic_icu.sofa`;
+CREATE TABLE `physionet-data.mimic_icu.sofa` AS
+with wt AS
+(
+  SELECT ie.stay_id
+    -- ensure weight is measured in kg
+    , avg(CASE
+        WHEN itemid IN (762, 763, 3723, 3580, 226512)
```

Comment:
any logic related to an `itemid` that is < 220000 can be deleted

Reply:
What do you mean by this? Are you saying we can delete this line of code?

Reply:
yep!